dev/io.openliberty.org.jboss.resteasy.common/src/io/openliberty/restfulWS/introspector/LibertyResteasyEndpointIntrospector.java

```diff
@@ -0,0 +1,93 @@
+/*******************************************************************************
+ * Copyright (c) 2021 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.restfulWS.introspector;
+
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.ConfigurationPolicy;
+
+import com.ibm.ws.container.service.app.deploy.ApplicationInfo;
+import com.ibm.ws.container.service.state.ApplicationStateListener;
+import com.ibm.ws.container.service.state.StateChangeException;
+import com.ibm.wsspi.logging.Introspector;
+
+@Component(service = { ApplicationStateListener.class, Introspector.class, LibertyResteasyEndpointLoggingIntrospector.class },
+           immediate = true,
+           configurationPolicy = ConfigurationPolicy.IGNORE,
+           property = "service.vendor=IBM")
+public class LibertyResteasyEndpointIntrospector implements ApplicationStateListener, Introspector, LibertyResteasyEndpointLoggingIntrospector {
+
+    private static final Map<String, List<String>> applicationEndpoints = new ConcurrentHashMap<String, List<String>>();
+
+    @Override
+    public void applicationStarting(ApplicationInfo appInfo) throws StateChangeException {
+        // add the application to the map
+        applicationEndpoints.put(appInfo.getName(), new ArrayList<String>());
+    }
+
+    @Override
+    public void applicationStarted(ApplicationInfo appInfo) throws StateChangeException {}
+
+    @Override
+    public void applicationStopping(ApplicationInfo appInfo) {}
+
+    @Override
+    public void applicationStopped(ApplicationInfo appInfo) {
+        // remove all endpoints for the application to prevent a memory leak
+        applicationEndpoints.remove(appInfo.getName());
+    }
+
+    @Override
+    public String getIntrospectorName() {
+        return "LibertyResteasyEndpointIntrospector";
+    }
+
+    @Override
+    public String getIntrospectorDescription() {
+        return "Log the JAX-RS endpoints for each Application.";
+    }
+
+    @Override
+    public void introspect(PrintWriter out) throws Exception {
+        for (String key : applicationEndpoints.keySet()) {
+            out.println(key + " endpoints:");
+            for (String ep : applicationEndpoints.get(key)) {
+                out.println(ep);
```

Comment:
```suggestion
                out.println("  " + ep);
```

I'm thinking it might help (visually) if we indent the endpoints.

Reply:
So, this `ep` string is already formatted when it's created during `addEndpoint()`

example output:
```
context endpoints:
- GET /context/context/uriinfo1 - com.ibm.ws.jaxrs.fat.context.ContextRestService#listQueryParamNames1()
- GET /context/context/request1 - com.ibm.ws.jaxrs.fat.context.ContextRestService#listRequest1()
- GET /context/context/config1 - com.ibm.ws.jaxrs.fat.context.ContextRestService#listConfiguration1()
```